modules/market/src/main/java/com/opengamma/strata/market/curve/definition/XCcyIborIborSwapCurveNode.java

```diff
@@ -115,8 +118,9 @@ public DatedCurveParameterMetadata metadata(LocalDate valuationDate) {
   @Override
   public SwapTrade trade(LocalDate valuationDate, MarketData marketData) {
     double marketQuote = marketData.getValue(spreadKey) + additionalSpread;
-    double fxRate = marketData.getValue(fxKey);
-    return template.toTrade(valuationDate, BuySell.BUY, 1, fxRate, marketQuote);
+    FxRate fxRate = marketData.getValue(fxKey);
+    double rate = fxRate.fxRate(fxKey.getPair());
```

Comment:
This needs to use the convention's currency pair to extract the data to ensure that the rate is the right way round.